File: sample/MyApplication/app/src/main/java/com/example/uemotoakira/myapplication/MainActivity.java
package com.example.uemotoakira.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn = findViewById(R.id.button1);
        btn.setText("hello");
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Button b = (Button) view;
                b.setText("こんちは");
            }
        });

    }
}

File: sample/Application09/app/src/main/java/com/example/uemotoakira/application09/MainActivity.kt
package com.example.uemotoakira.application09

import android.app.AlarmManager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.app.PendingIntent
import android.content.Intent
import android.widget.Button


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val WAIT_MILLISEC = (10 * 1000).toLong()
        val alarmManager = getSystemService(ALARM_SERVICE) as AlarmManager

        /**
         * getActivity:::Activityを実行するためのPendingIntentを取得する
         */
        findViewById<Button>(R.id.button1).setOnClickListener {
            val pendingIntent1: PendingIntent = Intent(this, SubActivity::class.java).let {
                                                    PendingIntent.getActivity(this, 0, it, 0)
                                                }
            alarmManager.setExact(AlarmManager.RTC_WAKEUP,System.currentTimeMillis() + WAIT_MILLISEC, pendingIntent1)
        }

        /**
         * getBroadcast:::BroadcastReciverを実行するためのPendingIntentを取得する
         */
        findViewById<Button>(R.id.button2).setOnClickListener {
            val pendingIntent2: PendingIntent = Intent(this, BcastReceiver::class.java).let {
                                                    PendingIntent.getBroadcast(this, 0, it, 0)
                                                }
            alarmManager.setExact(AlarmManager.RTC_WAKEUP,System.currentTimeMillis() + WAIT_MILLISEC, pendingIntent2)
        }

    }
}
